### DISCUSSION TITLE
Is it possible to deploy serverless functinons using git instead of the vercel cli?

### DISCUSSION THREAD

**Discussion by btakita** [vercel/vercel#8507] [General]
When I run `vercel`, I'm presented with a 2GB upload. To save time & bandwidth, I'd like to deploy using git, just like a static site. Is this possible? If so how?

**Comment by amyegan** [comment]
Hi @btakita! You should be able to use either the [Vercel for GitHub](https://vercel.com/docs/concepts/git/vercel-for-github) integration or customize your own [GitHub Actions for Vercel Deployment](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel). Similar options are available for BitBucket and GitLab. Deciding which will work best depends on your app and what tools you are using.

If you need more guidance on configuration options, please share more details about your app setup. I would be happy to help!

**Reply by btakita** [reply]
Thank you. I see a section to create serverless fuctions, but don't see how to use serverless functions with static files. Are there any examples of a `vercel.json` for an app with serverless functions with a static file build?

**Reply by btakita** [reply]
I found https://github.com/vercel/examples/ which utilizes the Build Output API https://vercel.com/docs/build-output-api/v3

Looks promising.

**Reply by btakita** [reply]
I set up a build output to `.vercel/output` & ran `vercel deploy --prebuilt` & got an error. The logs read:

```
2022-09-03T15:54:42.618Z  Retrieving list of deployment files...
2022-09-03T15:54:42.875Z
2022-09-03T15:54:46.302Z  Downloading 13 deployment files...
2022-09-03T15:54:47.086Z  Using prebuilt build artifacts...
2022-09-03T15:54:47.096Z  [Error: EISDIR: illegal operation on a directory, read] {
2022-09-03T15:54:47.096Z    errno: -21,
2022-09-03T15:54:47.096Z    code: 'EISDIR',
2022-09-03T15:54:47.096Z    syscall: 'read'
2022-09-03T15:54:47.096Z  }
2022-09-03T15:54:47.820Z
```

Unfortunately, the logs do not say what directory is being read or any other debugging clues, leaving me in a position of trial & error. Please add more detailed logging, such as what files & directories are being accessed.

**Reply by amyegan** [reply]
Thanks for sharing this @btakita! Are you able to share your test repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? I'll need to reproduce the error to identify what logging changes would need to be made for this.
